### ISSUE TITLE
No Audio on Headphones, Selecting audio output device freezes Dota Client.

### ISSUE THREAD

**Issue by sugoidesune** [ValveSoftware/Dota2-Gameplay#12416]
### Description

Steps to reproduce:

Start Dota without bluetooth headphones connected to PC.
Audio works through wired speaker.
Connect bluetooth headphones to PC.
Regular PC audio works through bluetooth headphones . Dota audio **CONTINUES** through speaker.
Manually selecting headphones in Dota Audio settings as output device freezes Dota client.
(Re)starting Dota while headphones are already connected works and Dota audio works through bluetooth headphones.

Important:
Once Dota Client has successfully played audio through headphones this behavior disappears and headphones are instantly recognized even when reconnecting bluetooth headphones. **UNTIL** next restart of PC, where this behavior starts again.

Dota Version: 5,864
PC: Windows 11

**Comment by yoyoinneverland** [comment]
Disable your speakers in the sound control panel.

**Comment by sugoidesune** [comment]
> Disable your speakers in the sound control panel.

that makes no sense since the issue is with dota not recognizing new audio outputs AFTER being launched.
NOT with audio outputs that are present during its launch.

**Comment by yoyoinneverland** [comment]
Yeah and this is not tech support. Obviously something is taking your interface hostage. I told you to disable the speakers to maybe see if that would default every single program to your headphones, but it seems you'd rather theorize and make a final conclusion by yourself, so have fun.

**Comment by minion12a** [comment]
If you want to play a game with your kids, [fall guys](https://fallguys3.com) is a good choice. You can choose from a huge number of fun and exciting games.

**Comment by gmackall** [comment]
I also get issues from either attaching a new audio output, or removing it. 

If I start dota with my headphones plugged in, I can freely change the audio output in the settings. But as soon as I unplug the headphones and then try to change the audio output my game freezes. 

My game also freezes if I try to quit the game after unplugging my headphones.

**Comment by omTTmo** [comment]
The crash also happens when changing audio device in the client. When losing bluetooth connection for some reason while client is running i have to restart it to get back audio.

**Comment by ffind** [comment]
@yoyoinneverland this obviously an issue. I got the same problem when playing music while playing dota. Other games works fine, only in dota even openning settings simetime freezes the game and all the time sound becomes cracky.

Mac m1

**Comment by Dante0719** [comment]
I am reporting a serious and highly reproducible audio device handling bug in Dota 2 on Windows.

The problem occurs specifically when using a Bluetooth headset. Dota 2 does not properly recover its audio output after the Bluetooth audio device is disconnected and then reconnected while the game is already running.

This is extremely frustrating because the Bluetooth headset itself is working normally, Windows recognizes it correctly, and other applications and games immediately restore audio after the headset reconnects. Dota 2, however, remains completely silent until the game is restarted.

Steps to reproduce
Connect a Redmi Bluetooth headset to the PC.
Make sure the headset is connected before launching Dota 2.
Launch Dota 2 and enter the game.
Confirm that Dota 2 audio is working normally through the Bluetooth headset.
While Dota 2 is still running, remove the earbuds and put them back into the charging case, causing the Bluetooth connection to be disconnected.
Take the earbuds out of the case again and allow them to reconnect to Windows.
Windows successfully reconnects the Bluetooth headset.
Other applications can immediately output audio through the Bluetooth headset.
However, Dota 2 produces no sound at all.
Restarting Dota 2 immediately restores the audio.
Another reproducible scenario

The same problem occurs if I launch Dota 2 while the Bluetooth headset is not connected.

In this situation:

Windows audio works normally through the speakers.
I connect the Bluetooth headset after Dota 2 has already started.
Windows successfully switches to the Bluetooth headset.
Other applications have normal audio through the headset.
Dota 2 remains completely silent.
Restarting Dota 2 while the Bluetooth headset is already connected makes Dota 2 audio work normally.
Important observations

This does NOT appear to be a hardware failure of the Bluetooth headset.

I have tested the same Bluetooth headset with other games, including:

Counter-Strike
Grand Theft Auto V
Red Dead Redemption 2
Cyberpunk 2077

In all of these games, I can disconnect the Bluetooth headset while the game is running, put the earbuds back into the charging case, reconnect them, and continue hearing game audio normally.

The problem occurs specifically in Dota 2.

Windows itself also recognizes the headset correctly after reconnection, and other applications can immediately play audio through it.

I have already configured Dota 2 to use the specific Redmi Bluetooth Stereo output device instead of the default audio device, but this does not resolve the problem.

The Bluetooth Hands-Free audio profile has also been disabled, so this is not related to Dota 2 switching between Stereo and Hands-Free modes.

Expected behavior

When a Bluetooth audio device is disconnected and subsequently reconnected while Dota 2 is running, Dota 2 should automatically detect that the audio endpoint has returned and resume audio playback through the reconnected device.

At the very least, Dota 2 should re-enumerate or reinitialize the audio output device when the previously selected device becomes available again.

Actual behavior

After the Bluetooth headset disconnects and reconnects:

Windows recognizes the headset normally.
Other applications have normal audio.
Dota 2 remains completely silent.
Changing the Windows default audio device does not reliably restore Dota 2 audio.
Restarting Dota 2 is required to restore sound.

This strongly suggests that Dota 2 is retaining a stale audio device/audio endpoint reference after the Bluetooth device disappears and is not correctly reinitializing the audio output when the device becomes available again.

Severity

This issue is particularly problematic because Bluetooth earbuds are designed to be frequently removed and reconnected during normal use.

Players should not be required to completely restart Dota 2 every time they temporarily remove their wireless earbuds.

This can also be especially disruptive during an ongoing match, because restarting Dota 2 is obviously not an acceptable workaround during gameplay.

I sincerely hope the Dota 2 team can investigate this as soon as possible, particularly the handling of Windows audio device enumeration and Bluetooth audio endpoint changes while the game is running.

There appears to be a related historical report regarding Bluetooth headphones and Dota 2 audio device detection (#12416), but the issue described here also occurs when the Bluetooth device was already working correctly and is subsequently disconnected and reconnected during gameplay.

Please investigate the audio device hot-plug/reconnection handling in Dota 2.

Thank you very much for your attention to this issue. I would greatly appreciate it if the Dota 2 team could investigate and fix this as soon as possible.
